### PR TITLE
MINOR: Remove spurious sleep in ConsumerCoordinatorTest

### PR DESCRIPTION
**Pull request by ijuma**
Follow up from here https://github.com/apache/kafka/pull/12349#discussion_r925138410

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by showuon — APPROVED** [review]
LGTM! Thanks.

**Comment by showuon** [comment]
Triggering another jenkins build to make sure it passes the tests.
https://ci-builds.apache.org/job/Kafka/job/kafka-pr/view/change-requests/job/PR-12500/2/

**Comment by aiquestion** [comment]
Ah, sorry for the delay. I think time.sleep() here is needed here to make the pollTimer timeout. If we removed this, will hang in the `coordinator.onJoinPrepare`

**Comment by showuon** [comment]
> Ah, sorry for the delay. I think time.sleep() here is needed here to make the pollTimer timeout. If we removed this, will hang in the coordinator.onJoinPrepare

@aiquestion , thanks for the explanation. In that case, don't you think we should make the timer timeout to 0? That is:
```java
Timer pollTimer = time.timer(0L);
// no need to make the timeout expired here
// time.sleep(150);
boolean res = coordinator.onJoinPrepare(pollTimer, generationId, memberId);
assertFalse(res);
```

If you think this is a good suggestion, could you open another PR to improve it? Thanks.

**Review by showuon — APPROVED** [review]
LGTM. Thanks, @ijuma !

**Comment by showuon** [comment]
Weird! Looks like the test still stuck. I'll take a look after my vacation.

File: clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java [hunk 1]
```diff
@@ -1348,7 +1348,6 @@ public void testOnJoinPrepareWithOffsetCommitShouldKeepJoinAfterRebalanceTimeout
             String memberId = "consumer-42";
 
             Timer pollTimer = time.timer(100L);
```
Comment (aiquestion) [inline clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java:1350]: i think we also need to change time to `Timer pollTimer = time.timer(0L);` if we are going to remove `time.sleep(150);` below.  
i think the test hang here.

**Pull request closed without merging**